sdk/src/main/java/com/google/cloud/dataflow/sdk/io/CompressedSource.java

```diff
@@ -424,7 +455,10 @@ protected final boolean readNextRecord() throws IOException {
      */
     @Override
     protected final long getCurrentOffset() {
-      return readerDelegate.getCurrentOffset();
+      if (splittable) {
```

Comment:
Here's my concern: the reader's currentOffset is unrelated to the offsets in the dimension that the source, its size, and the way that the service reads them. The reader is getting a decompressed stream, so the source could be 100B big and the currentOffset could go to 100M.

Maybe this is not a bug -- progress will just quickly go to 100% and stay there? -- but seemed undesirable.